cupyx/scipy/ndimage/filters.py

```diff
@@ -76,40 +164,105 @@ def _get_output(output, input, shape=None):
     return output
 
 
-def _correlate_or_convolve(input, weights, output, mode, cval, origin,
-                           convolution):
-    if input.dtype.kind == 'c':
-        raise TypeError('Complex type not supported.')
-    if not hasattr(origin, '__getitem__'):
-        origin = [origin, ] * input.ndim
+def _fix_sequence_arg(arg, ndim, name, conv=lambda x: x):
+    if hasattr(arg, '__iter__') and not isinstance(arg, str):
+        lst = [conv(x) for x in arg]
+        if len(lst) != ndim:
+            msg = "{} must have length equal to input rank".format(name)
+            raise RuntimeError(msg)
     else:
-        origin = list(origin)
-    wshape = [ii for ii in weights.shape if ii > 0]
-    if len(wshape) != input.ndim:
-        raise RuntimeError('filter weights array has incorrect shape.')
-    if convolution:
-        weights = weights[tuple([slice(None, None, -1)] * weights.ndim)]
-        for ii in range(len(origin)):
-            origin[ii] = -origin[ii]
-            if weights.shape[ii] % 2 == 0:
-                origin[ii] -= 1
-    for _origin, lenw in zip(origin, wshape):
-        if (lenw // 2 + _origin < 0) or (lenw // 2 + _origin >= lenw):
-            raise ValueError('invalid origin')
+        lst = [conv(arg)] * ndim
+    return lst
+
+
+def _check_origin(origin, width):
+    origin = int(origin)
+    if (width // 2 + origin < 0) or (width // 2 + origin >= width):
+        raise ValueError('invalid origin')
+    return origin
+
+
+def _check_mode(mode):
     if mode not in ('reflect', 'constant', 'nearest', 'mirror', 'wrap'):
         msg = 'boundary mode not supported (actual: {}).'.format(mode)
         raise RuntimeError(msg)
+    return mode
+
+
+def _check_axis(axis, ndim):
+    axis = int(axis)
+    if axis < 0:
+        axis += ndim
+    if axis < 0 or axis >= ndim:
+        raise ValueError('invalid axis')
+    return axis
+
+
+def _convert_1d_args(ndim, weights, origin, axis):
+    if weights.ndim != 1 or weights.size < 1:
+        raise RuntimeError('incorrect filter size')
+    axis = _check_axis(axis, ndim)
```

Comment:
use `cupy.util._normalize_axis_index` as it's essentially the same code

Reply:
@emcastillo Are you referring to function `_check_axis`? To be replaced by `cupy.util._normalize_axis_index`

Reply:
Yes! if possible :)

Reply:
Completed in #3390

Reply:
Thanks!